### ISSUE TITLE
Documentation for new Tax-Calculator user through CLI? 

### ISSUE THREAD

**Issue by MattHJensen** [open-source-economics/Tax-Calculator#1163]
Is there any documentation for a beginning Tax-Calculator user who would like to get started with Tax-Calculator from the command line interface using the latest Conda version of Tax-Calculator rather than the source code from GitHub? Am I right to think that this user wouldn't be using Python or git directly?

**Comment by martinholmer** [comment]
@MattHJensen said:
> Is there any documentation for a beginning Tax-Calculator user who would like to get started with Tax-Calculator from the command line interface using the latest Conda version of Tax-Calculator rather than the source code from GitHub?

No, there is no such documentation as far as I know.  The reason there isn't is that there is very little that can be done with the Conda version of Tax-Calculator because it doesn't include the private `puf.csv` file.  About all that can be done with the Conda version is to prepare CSV-formatted input files (with one or more made-up tax filing units) and process them at the command-line like this:
```
$ python inctax.py my-data.csv MY-DATA-YEAR [--reform REFORMFILE [--assump ASSUMPFILE]]
```
The resulting output file would contain for each filing unit the 28 variables described [here](https://github.com/open-source-economics/Tax-Calculator/blob/master/taxcalc/validation/CSV_OUTPUT_VARS.md).

My thought was that there would be more demand for such documentation after the `cps.csv` file --- and the associated `cps-weights.csv` and `cps-factors.csv` files --- are included in the Conda version of Tax-Calculator.

@MattHJensen also asked:
> Am I right to think that this user wouldn't be using Python or git directly?

Yes, such users would not need git (because they would be uninterested in making changes to the source code), but they would need a Python interpreter on their computer in order to execute the Tax-Calculator via the `inctax.py` command-line interface to Tax-Calculator.

**Comment by martinholmer** [comment]
@MattHJensen asked:
> @feenberg expressed interest in such an interface, and he does have access to the `puf.csv` file. Could he [do the following]?
> ```
> python inctax.py puf.csv 2022 --blowup --weights [--reform REFORMFILE [--assump ASSUMPFILE]]
> ```

Yes, of course, that is what I do every day.  The `--weights` option adds to the 28-TAXSIM-variable output a variable 29 with is the weight in the specified year.  Then you can do anything you want with the space-delimited output file: process it with AWK, process it with SAS, import it into a spreadsheet, etc.

**Comment by martinholmer** [comment]
I forgot to mention in issue #1163 that the `inctax.py` command-line interface to Tax-Calculator has extensive documentation that can be shown on the screen by entering the following at the operating-system command prompt (in this example $):
```
$ python inctax.py --help
```

@MattHJensen @feenberg

**Comment by martinholmer** [comment]
An afterthought on issue #1163:

I actually have very little experience using the Conda taxcalc package, so my earlier responses to the question posed by @MattHJensen may not be completely accurate.  The key question is whether or not the `inctax.py` file (which is located in the top-level directory above the taxcalc directory) is included in the Conda taxcalc package.  I don't know the answer to that question.  If it is included, then everything I said in #1163 is true.  But if `inctax.py` is not included in the taxcalc Conda package, then an extra installation step would be needed to get that file.

Can anybody with more experience with the taxcalc Conda package answer this question?

@MattHJensen @feenberg @PeterDSteinberg

**Comment by PeterDSteinberg** [comment]
@martinholmer @MattHJensen @feenberg - I agree that the current placement of `inctax.py` at the top level of the directory is inconsistent with usage of `taxcalc` from conda installation.  The simple fix would be for me to move `inctax.py` into the `taxcalc` dir, then in the `setup.py` create a `entry_points` - `console_scripts` argument to the `setup` function.  [This setuptools documentation](http://setuptools.readthedocs.io/en/latest/setuptools.html#automatic-script-creation) shows how `entry_points` works.  In the example on that page, they create command line interfaces `foo` and `bar` which are just aliases to importable Python functions.  In our case, where they have `foo = my_package.some_module:main_func`, we would have `inctax = taxcalc.inctax:main`, allowing us to do the following at command prompt:

```
source activate taxcalc-dev
inctax --help
```
And it would be the same as running what is currently `python inctax.py --help`.  The snippet above would work the same in the case 1) user has installed from source, 2) user installed with conda.

If this sounds good to you all, I can make the changes - probably less than an hour of work.

**Comment by martinholmer** [comment]
I don't like the #1163 solution suggested by @PeterDSteinberg because it requires (even casual) users to learn about and and use the (unknown and complex, to them) conda environment apparatus.  If you want an effective way to **discourage** users --- I'm not talking about developer, here --- from getting started with Tax-Calculator, then requiring them to use the conda environment would be the perfect way to keep them away.  Let's let this issue be for a while in order to give me some time to come up with an alternative.  I've been thinking about alternatives for a while since @MattHJensen assigned this task to me.

@feenberg

**Comment by MattHJensen** [comment]
Just a point of clarification  -- in @PeterDSteinberg's suggestion, would the taxcalc-dev environment need to be activated if the user installed taxcalc with conda?

**Comment by PeterDSteinberg** [comment]
@MattHJensen - The conda environment would not *necessarily* have to be activated for the console script entry point, *if* they have all the requirements installed.  The only requirement for getting the console script entry point working is to do `python setup.py install` or `python setup.py develop` (which is in effect done by `conda install` for you).  Typically, however, it would probably be easiest if you were installing from source to do:

`conda env create && source activate taxcalc-dev && python setup.py install`

before trying the console script entry point.

**Comment by MattHJensen** [comment]
@PeterDSteinberg, I'm actually interested in the case where the user is _not_ installing from source, and instead `conda installs` the latest versioned release of `taxcalc`.  

The reason I am clarifying this is that we already expect our developers (and others installing from source) to activate the `taxcalc-dev` environment, while we don't currently expect our `conda install` users to activate the environment. So if your proposed solution does not require `conda install` users to activate a conda environment, then it doesn't actually require anyone _new_ to use conda environments, and @martinholmer's initial concern may be misplaced -- assuming I understood it correctly. 

This is not to say, of course, that we shouldn't be supportive of alternative solutions to the issue at hand, but I think clarification of this point could be helpful.

**Comment by PeterDSteinberg** [comment]
@MattHJensen - Your summary is correct: I did not mean to imply that the same user/machine would do both `conda install taxcalc` and `source activate taxcalc-dev`.  I think of these two options as alternate ideas, with `source activate taxcalc-dev` being associated with running `python setup.py install` after activating the environment.

**Comment by martinholmer** [comment]
Documentation on how to **use** Tax-Calculator is currently being developed.  When completed, I believe that documentation will resolve issue #1163.

**Comment by martinholmer** [comment]
The initial effort to prepare a document on using Tax-Calculator from the operating system command line (without any Python programming) has been completed and can be read in your browser at [this link](http://open-source-economics.github.io/Tax-Calculator/), which has the following URL:
```
http://open-source-economics.github.io/Tax-Calculator/
```
The completion of an initial draft of the user documentation resolves issue #1163, which is now being closed.
